creator-node/src/fileManager.js

```diff
@@ -95,57 +96,105 @@ async function saveFileToIPFSFromFS (req, srcPath, fileType, sourceFile, transac
  *    - If multihash not available locally, fetch file from IPFS.
- *  - Write file to disk.
+ *  - If file is not available via IPFS try other cnode gateways for user's replica set.
  *  - Add file to local inode if not already.
  */
-async function saveFileForMultihash (req, multihash, expectedStoragePath) {
+async function saveFileForMultihash (req, multihash, expectedStoragePath, gatewaysToTry) {
   // If file already stored on disk, return immediately.
   if (fs.existsSync(expectedStoragePath)) {
     req.logger.info(`File already stored at ${expectedStoragePath} for ${multihash}`)
     return expectedStoragePath
   }
 
   // If file not already stored, fetch from IPFS and store at storagePath.
-  const ipfs = req.app.get('ipfsAPI')
   let fileBuffer = null
-  req.logger.info(`Storing file at ${expectedStoragePath} for track multihash ${multihash}`)
 
   // If multihash already available on local INode, cat file from local ipfs node
-  req.logger.info(`checking if ${multihash} already available on local ipfs node`)
+  req.logger.debug(`checking if ${multihash} already available on local ipfs node`)
   try {
-    fileBuffer = await ipfs.cat(multihash)
-    req.logger.info(`Retrieved file for ${multihash} from local ipfs node`)
+    fileBuffer = await Utils.ipfsCat(multihash, req, 500)
+    req.logger.debug(`Retrieved file for ${multihash} from local ipfs node`)
+    // Write file to disk.
+    await writeFile(expectedStoragePath, fileBuffer)
+    req.logger.info(`wrote file to ${expectedStoragePath}, obtained via ipfs cat`)
   } catch (e) {
     req.logger.info(`Multihash ${multihash} is not available on local ipfs node`)
   }
 
   // If file not already available on local INode, fetch from IPFS.
   if (fileBuffer === null) {
-    req.logger.info(`Attempting to get ${multihash} from IPFS`)
+    req.logger.debug(`Attempting to get ${multihash} from IPFS`)
     let output
     try {
-      output = await ipfs.get(multihash)
+      // ipfsGet returns a BufferList object, not a buffer
+      // but still compatible into writeFile directly
+      output = await Utils.ipfsGet(multihash, req, 5000)
+      fileBuffer = output
+      req.logger.debug(`retrieved file for multihash ${multihash} from local ipfs node`)
+
+      // Write file to disk.      
+      const destinationStream = fs.createWriteStream(expectedStoragePath)
+      await new Promise((resolve, reject) => {
+        fileBuffer.pipe(destinationStream)
+        fileBuffer.on('end', () => { resolve() })
+        fileBuffer.on('error', err => { destinationStream.end(); reject(err) })
+      })
+      req.logger.info(`wrote file to ${expectedStoragePath}, obtained via ipfs get`)
     } catch (e) {
-      throw new Error(`Failed to retrieve file for multihash ${multihash} from IPFS`)
+      req.logger.info(`Failed to retrieve file for multihash ${multihash} from IPFS ${e.message}`)
     }
-    if (output.length !== 1) throw new Error('Audius track segment multihash must map to 1 file')
```

Comment:
do we still need this check?

Reply:
nope we don't. output is totally different now